sql/mdl.cc

```diff
@@ -304,6 +357,87 @@ bool Deadlock_detection_visitor::enter_node(MDL_context *node)
   return m_found_deadlock;
 }
 
+#if MARIA_PLUGIN_INTERFACE_VERSION < 0x0200
```

Comment:
Why?

Reply:
this copied from sql_class.cc
the code is

=========
#if MARIA_PLUGIN_INTERFACE_VERSION < 0x0200
/**                                                                                                 
  TODO: This function is for API compatibility, remove it eventually.                               
  All engines should switch to use thd_get_error_context_description()                              
  plugin service function.                                                                          
*/
extern "C"
char *thd_security_context(THD *thd,
                           char *buffer, unsigned int length,
                           unsigned int max_query_len)
{
  return thd_get_error_context_description(thd, buffer, length, max_query_len);
}
#endif
=========

If the macro MARIA_PLUGIN_INTERFACE_VERSION's value change to equal or bigger, this will cause a compile error.

Is it better to remove this macro from this patch?